src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -49,17 +46,18 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
   <nav>
     <a id="navTitle" href="/">CodeU Chat App</a>
     <a href="/conversations">Conversations</a>
-      <% if (request.getSession().getAttribute("user") != null) { %>
-    <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
-    <a href="/profile">Profile</a>
-    <% } else { %>
+    <% if(request.getSession().getAttribute("user") != null){ %>
+      <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
+      <a href="/profile/<%=request.getSession().getAttribute("user")%>">Profile</a>
+    <% } else{ %>
       <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
-    <a href="/admin">Admin</a>
+    <a href="/admin.jsp">Admin</a>
     <a href="/activityfeed">Activity Feed</a>
```

Comment:
Sometimes adding ".jsp" to the admin file doesn't work often. Have you tried running it locally?